build/repo.targets

```diff
@@ -78,4 +78,42 @@
       <Output TaskParameter="ExecutablePath" PropertyName="DotNetPath" />
     </GetDotNetHost>
   </Target>
+
+  <Target Name="_UpdateDependenciesFileInKorebuild">
```

Comment:
As you might have guessed, this PR only handles the part of generating the dependencies used for a particular build, but there is still work to do to wire this up in Korebuild and our updating our repos + handling in Universe build scenarios